includes/razorpay-subscriptions.php

```diff
@@ -0,0 +1,372 @@
+
+        return array($planId, true);
+    }
+
+    protected function createPlan($planArgs)
+    {
+        try
+        {
+            $plan = $this->api->plan->create($planArgs);
+        }
+        catch (Exception $e)
+        {
+            $message = $e->getMessage();
+
+            throw new Errors\Error(
+                $message,
+                WooErrors\ErrorCode::API_PLAN_CREATION_FAILED,
+                400
+            );
+        }
+
+        // Storing the plan id as product metadata, unique set to true
+        return $plan['id'];
+    }
+
+    protected function getPlanArguments($product)
+    {
+        $productId = $product['product_id'];
+
+        $period       = WC_Subscriptions_Product::get_period($productId);
+        $interval     = WC_Subscriptions_Product::get_interval($productId);
+        $salePrice    = get_post_meta($productId)['_sale_price'][0];
+        $recurringFee = WC_Subscriptions_Product::get_price($productId);
+
+        //
+        // If the item is on sale, sell it for the sale price
+        //
+        if (empty($salePrice) === false)
+        {
+            $recurringFee = $salePrice;
+        }
+
+        //
+        // Ad-Hoc code
+        //
+        if ($period === 'year')
+        {
+            $period = 'month';
+
+            $interval *= 12;
+        }
+
+        $planArgs = array(
+            'period'   => $this->getProductPeriod($period),
+            'interval' => $interval
+        );
+
+        $item = array(
+            'name'     => $product['name'],
+            'amount'   => (int) round($recurringFee * 100),
+            'currency' => get_woocommerce_currency(),
+        );
+
+        if ($item['currency'] !== self::INR)
+        {
+            $this->razorpay->handleCurrencyConversion($item);
+        }
+
+        $planArgs['item'] = $item;
+
+        return $planArgs;
+    }
+
+    public function getDisplayAmount($orderId)
+    {
+        $order = new WC_Order($orderId);
+
+        $product = $this->getProduct($order);
+
+        $productId = $product['product_id'];
+
+        $recurringFee = WC_Subscriptions_Product::get_price($productId);
+
+        $salePrice = get_post_meta($productId)['_sale_price'][0];
+
+        //
+        // If the item is on sale, sell it for the sale price
+        //
+        if (empty($salePrice) === false)
+        {
+            $recurringFee = $salePrice;
+        }
+
+        $signUpFee = WC_Subscriptions_Product::get_sign_up_fee($productId);
+
+        return $recurringFee + $signUpFee;
+    }
+
+    protected function getProductPeriod($period)
+    {
+        $periodMap = array(
+            'day'   => 'daily',
+            'week'  => 'weekly',
+            'month' => 'monthly',
+            'year'  => 'yearly'
+        );
+
+        return $periodMap[$period];
+    }
+
+    protected function getCustomerId($order)
+    {
+        $data = $this->getCustomerInfo($order);
+        $data['fail_existing'] = '0';
```

Comment:
Give a link to the [docs](https://docs.razorpay.com/v1/page/customers-api) so it is clear what this does.